aioamqp/tests/test_basic.py

```diff
@@ -158,3 +158,24 @@ def qcallback(channel, body, envelope, properties):
         envelope = yield from qfuture
 
         yield from self.channel.basic_reject(envelope.delivery_tag)
+
+    @testing.coroutine
+    def test_basic_nack(self):
+        queue_name = 'queue_name'
+        exchange_name = 'exchange_name'
+        routing_key = ''
+        yield from self.channel.queue_declare(queue_name, exclusive=False, no_wait=False)
+        yield from self.channel.exchange_declare(exchange_name, type_name='fanout')
+        yield from self.channel.queue_bind(queue_name, exchange_name, routing_key=routing_key)
+        yield from self.channel.publish("payload", exchange_name, queue_name)
+
+        qfuture = asyncio.Future(loop=self.loop)
+
+        @asyncio.coroutine
+        def qcallback(channel, body, envelope, properties):
+            qfuture.set_result(envelope)
+
+        yield from self.channel.basic_consume(qcallback, queue_name=queue_name)
+        envelope = yield from qfuture
+
+        yield from self.channel.basic_client_nack(envelope.delivery_tag)
```

Comment:
You can also nack this inside you `qcallback()` but you sill want to know if your callback is correctly called :p